compiler/parser/errors/errors.go

```diff
@@ -41,6 +42,21 @@ func (e *Error) IsUnexpectedEnd() bool {
 	return e.ErrType == UnexpectedEndError
 }
 
+// IsUnexpectedTok checks if error is unexpected token error
+func (e *Error) IsUnexpectedToken() bool {
+	return e.ErrType == UnexpectedTokenError
+}
+
+// IsUnexpectedCase checks if error is unexpected token error for 'case'atement
+func (e *Error) IsUnexpectedCase() bool {
+	return e.ErrType == UnexpectedTokenError && len(e.Message) >= 49 && e.Message[0:49] == "expected next token to be WHEN, got EOF() instead"
```

Comment:
I think what we should do is to include the wrong token in the `Error`. since most of the parser error exists in the expected token form

Reply:
Makes sense!
I found some more minor issues in the PR and will update soon.

Reply:
fixed